library/src/scala/Tuple.scala

```diff
@@ -74,6 +79,12 @@ object Tuple {
     case h *: t => F[h] *: Map[t, F]
   }
 
+  type Zip[T1 <: Tuple, T2 <: Tuple] <: Tuple = (T1, T2) match {
+    case (h1 *: t1, h2 *: t2) => (h1, h2) *: Zip[t1, t2]
+    case (Unit, _) => Unit
+    case (_, Unit) => Unit
```

Comment:
Probably missing `case _ => Tuple`

```scala
val t1: Int *: Long *: Tuple = ???
val t2: Int *: Char *: Tuple = ???
val t3: (Int, Int) *: (Long, Char) *: Tuple = t1.zip(t2)
```

But maybe the `<: Tuple` is enough.

Reply:
Why would that case be necessary? I think the type bound is enough to specify that the result is a tuple

Reply:
I'll add a test based on your snippet and add that case if it misbehaves

Reply:
Also check how the error message looks.

Reply:
Looks like your test cases passes without that clause.

Reply:
OK

Reply:
I would add it for completeness and to giving it a chance of simplifying the type in the error messages.

Reply:
It is already in the test suite – see the "More Zip" comment in `tuple-ops.scala`.

Reply:
> I would add it for completeness and to giving it a chance of simplifying the type in the error messages.

Can you clarify?

Reply:
To reduce `Zip[Tuple, Tuple]` to just `Tuple`.